mmv1/third_party/terraform/fwprovider/framework_provider_access_token_test.go

```diff
@@ -0,0 +1,254 @@
+package fwprovider_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+// TestAccFwProvider_access_token is a series of acc tests asserting how the plugin-framework provider handles access_token arguments
+// It is plugin-framework specific because the HCL used provisions plugin-framework-implemented resources
+// It is a counterpart to TestAccSdkProvider_access_token
+func TestAccFwProvider_access_token(t *testing.T) {
+	testCases := map[string]func(t *testing.T){
+		"config takes precedence over environment variables":                                                        testAccFwProvider_access_token_configPrecedenceOverEnvironmentVariables,
+		"when access_token is unset in the config, environment variables are used in a given order":                 testAccFwProvider_access_token_precedenceOrderEnvironmentVariables, // GOOGLE_OAUTH_ACCESS_TOKEN
+		"when access_token is set to an empty string in the config the value isn't ignored and results in an error": testAccFwProvider_access_token_emptyStringValidation,
+		"access_token conflicts with credentials":                                                                   testAccFwProvider_access_token_conflictsWithCredentials,
```

Comment:
Are there other test cases that would make sense here?

Reply:
I think we need a test where we configure the provider with an auth token and assert it can provision stuff with that auth.

We'd need the acc test to A) have a provider that's given credentials from CI, B) uses google_client_config to get a token from those creds, C) configures an aliased provider with the token, and D) asserts that the aliased provider can provision something ok with that token.

I think there are some similar tests that should be implemented to test google_client_config (different creds and failure modes) but should be kept out of this PR/provider level acc tests

Reply:
Update: Using google_client_config to facilitate this is way too much of a pain, instead I use some code to supply the config with an access token directly

Reply:
Addressed in 4b20113 and 6a3e1a7